Some suggested corrections and modifications for increased efficiency

jankowtf opened the pull request:
Hi Richie,

nice little package!

It's the first time that I actually contribute via git/GitHub, so please bare with me in case I did anything "wrong" ;-) 

Best regards,
Janko

Commit:
Update paths.R

IMHO, `recompose_path.decomposed_path()` should return an object identical to the input `x` of `decompose_path()`. 

My patch makes sure that 
- trailing slashes for directory names, as you call it (I would call them *directory paths* as for me a *directory name* would only be the *basename* of a directory path; I would make the same distinction for *file paths* and *file names*), are removed
- `""` is the result for components with both empty directory names and file extensions

Commit:
* R/paths.R/decompose_path:
  - corrected code in for empty 'x'
  - also corrected relevant unit test: /inst/tests/test_decompose_path.R
* R/paths.R/recompose_path:
  - corrected code in order to get identical (or close to identical)
    results to the input 'x' of 'decompose_path' when applying the
    function
  - added unit test: /inst/tests/test_recompose_path.R
* R/paths.R/copy_path:
  - rewrite of the original function in order to make it more efficient
  - the old version has been renamed to 'copy_path_old'
  - added unit test: /inst/tests/test_copy_path.R; note that it will fail
    for tests of 'copy_path_old' as IMHO the function currently does not
    handle 'mapply' scenarios correctly (i.e. multiple sources and/or
    multiple targets)
  - added a profiling script: /inst/prof/copy_path.R; the script
    illustrates that 'copy_path_old' is roughly twice as slow as
    'copy_path'

jankowtf commented:
Oh, sorry: you can disregard commit `4488eaa`, that came a bit fast ;-)

richierocks commented:
Janko,

Thanks for your input.  There's a lot of stuff here, and a lot of it looks
good.  Since it's a bundle of things and my git skills are rather
rudimentary, I think it will be easiest to go through it and manually cut
and paste things into the existing codebase instead of accepting the whole
pull request and trying to merge things.

So if I reject the pull request, it isn't because I'm rejecting your code,
it's just that I suck at version control.  Your code will make into the
package, and I'll credit you in the commit comments and in the package
description.

On 3 May 2014 02:54, Janko Thyson notifications@github.com wrote:

> Oh, sorry: you can disregard commit 4488eaa, that came a bit fast ;-)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/richierocks/pathological/pull/1#issuecomment-42089919
> .

## 

Regards,
Richie

Learning R http://shop.oreilly.com/product/0636920028352.do
4dpiecharts.com

jankowtf commented:
No worries, man. 

First of: your package, your rules ;-) Just process my stuff the way you're most comfortable with.

Second, I can totally relate: while I really like git and its version control philosophy, I sometimes still have trouble mastering the nitty gritty details ;-) Hence, I also wouldn't know how to selectively accept only certain parts of a pull request.

Thanks for the credit, I appreciate it.

Best regards,
Janko

Commit:
rev2

Commit:
rev3

Commit:
fixed conflicts

Commit:
* /R/paths.R:
  - changed own version of 'dir_copy' to 'dir_copy_rappster'; 'dir_copy'
    corresponds to original version again
  - changed own version of 'recompose_path' to 'recompose_path_rappster';
    'recompose_path' corresponds to original version again
  - some minor merge conflicts fixed:
